osu.Game/Rulesets/Objects/Drawables/DrawableHitObject.cs

```diff
@@ -333,6 +329,8 @@ protected sealed override void OnFree(HitObjectLifetimeEntry entry)
             Entry.NestedEntries.RemoveAll(nestedEntry => nestedEntry is SyntheticHitObjectEntry);
             ClearNestedHitObjects();
 
+            // Changes in state trigger defaults applied trigger state updates.
```

Comment:
This sounds circular. What are the "changes in state" at the start there? Hitobject property changes? Not literal changes to `DrawableHitObject.State`?

Reply:
The "changes in state" refers to state of the `HitObject`

Reply:
Updated comment to be clearer in https://github.com/ppy/osu/pull/24508/commits/5be533578499d25c54e18ea76ec2aae2a32a9b24.